consensus/service/src/byzantine_ledger.rs

```diff
@@ -519,7 +521,7 @@ impl<
                 log::info!(self.logger, "sync_service reports we are no longer behind!");
 
                 // Reset scp state.
-                self.scp.clear_pending_slots();
+                self.scp.reset_slot_index(self.cur_slot);
```

Comment:
Have you tested catchup? I believe this is wrong since `self.cur_slot` points to the old current slot, before catchup has completed.

Reply:
Good catch. I changed this to reset the scp state after cur_slot has been updated, and so far catch-up seems to work.